extension/history.js

```diff
@@ -20,107 +20,131 @@ export class Database {
         const database = e.target.result;
         resolve(database);
       };
-      request.onerror = function(e) {
-        reject("Error: failed to create a database!");
+      request.onerror = e => {
+        const code = e.target.errorCode;
+        reject(new Error(`Failed to create a database: ${code}`));
       };
     });
   }
 
   get(primaryKey, TBName) {
```

Comment:
TBName should probably be changed to tbName if we're following other conventions